### PR TITLE
Add clean `--untracked`

### LINKED ISSUE #4831: opam clean --untracked
We currently do not have a way to detect and clean (interactively) a switch containing external/modified files.
It would be nice to have this option around so that when we get warnings such as:
```
[NOTE] While removing ocaml-base-compiler.4.12.0: not removing non-empty directories:
  - share
```
we could do something about it and know what was added or modified in this directory.

Voiced in https://github.com/ocaml/opam/issues/4781

### PR DESCRIPTION
**Pull request by rjbou**
fix #4831
See test file for behaviour

### CODE REVIEW

File: src/client/opamCommands.ml [hunk 1]
```diff
@@ -3663,7 +3667,7 @@ let clean cli =
     in
     let switches =
       if all_switches then OpamGlobalState.switches gt
-      else if switch then match OpamStateConfig.get_switch_opt () with
+      else if switch || untracked then match OpamStateConfig.get_switch_opt () with
```
Comment (kit-ty-kate) [inline src/client/opamCommands.ml:3670]: One could call `opam clean --untracked --all-switches`
```suggestion
      else if switch then match OpamStateConfig.get_switch_opt () with
```

File: tests/reftests/clean.test [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  - ${BASEDIR}/OPAM/clean/lib/things/second
+Remove directory? [Y/n] y
+Cleaning up switch clean
+### opam install things
+The following actions will be performed:
+  - install things 1
+
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed things.1
+Done.
```
Comment (kit-ty-kate) [inline tests/reftests/clean.test:91]: Could you add something like:
```
### <modify an installed file>
### opam clean --untracked
<see if it’s actually removed>
### opam clean --untracked
<see if nothing is actually done>
```

File: src/client/opamCommands.ml (hunk 1 above)
Comment (rjbou) [reply src/client/opamCommands.ml:3670]: It is already handled by the line above
```
       if all_switches then OpamGlobalState.switches gt 
```

Comment (kit-ty-kate) [reply src/client/opamCommands.ml:3670]: Ah my bad

File: tests/reftests/clean.test (hunk 2 above)
Comment (rjbou) [reply tests/reftests/clean.test:91]: with an installed or not installed package ?

Comment (kit-ty-kate) [reply tests/reftests/clean.test:91]: why not both, just in case

Comment (rjbou) [reply tests/reftests/clean.test:91]: ok

**Comment by AltGr** [comment]
Nice! I have a few comments:
- the code would probably deserve to be in its own function
- the interactive mode is likely to spam a *lot* of questions... maybe list them and ask once ? Otherwise require `--dry-run`/`--yes` a bit like `git clean -n/f`, maybe ?
- do you make sure to minimise the tree, e.g. if there is a whole subtree that is expected to be removed, ask just once for its root (or if you want to list everything, make sure to list the contents first, then the dir — if empty — with an `rmdir`)
- `OpamConsole.msg "Cleaning up switch %s\n" (OpamSwitch.to_string sw);` should probably be above the code

**Comment by rjbou** [comment]
Updated!
```
### opam clean --untracked
Cleaning up switch clean
Remaining directories and files:
  - ${BASEDIR}/OPAM/clean/bin/first
  - ${BASEDIR}/OPAM/clean/lib/things/second
  - ${BASEDIR}/OPAM/clean/lib/things

Remove them? [Y/n] y

**Review by kit-ty-kate — APPROVED** [review]

**Pull request merged**